orderer/consensus/etcdraft/blockpuller.go

```diff
@@ -151,72 +150,63 @@ func NewBlockFetcher(support consensus.ConsenterSupport,
 	// TODO: change this to use the new mapping of consenters in the channel config
 	// To tolerate byzantine behaviour of `f` faulty nodes, we need a total of `3f + 1` nodes.
 	// f = maxByzantineNodes, total = len(endpoints)
-	maxByzantineNodes := uint64(len(endpoints)-1) / 3
-
-	// timeout for time based shuffling
-	shuffleTimeout := shuffleTimeoutMultiplier * clusterConfig.ReplicationPullTimeout
-	shuffleTimeoutThrehold := shuffleTimeoutPercentage
+	maxByzantineNodes := (len(endpoints) - 1) / 3
 
 	fc := cluster.FetcherConfig{
-		ShuffleTimeout:               shuffleTimeout,
-		CensorshipSuspicionThreshold: time.Minute,
-		MaxByzantineNodes:            maxByzantineNodes,
 		Channel:                      support.ChannelID(),
 		TLSCert:                      der.Bytes,
 		Endpoints:                    endpoints,
 		FetchTimeout:                 clusterConfig.ReplicationPullTimeout,
+		CensorshipSuspicionThreshold: time.Duration((int64(clusterConfig.ReplicationPullTimeout) * shuffleTimeoutPercentage / 100)),
+		PeriodicalShuffleInterval:    shuffleTimeoutMultiplier * clusterConfig.ReplicationPullTimeout,
+		MaxRetries:                   uint64(clusterConfig.ReplicationMaxRetries),
+		MaxByzantineNodes:            maxByzantineNodes,
+	}
+
+	// TODO: Should this func be moved to `orderer/common/cluster` package
+	blockVerifierFunc := func(header *common.BlockHeader, metadata *common.BlockMetadata) error {
```

Comment:
I'll make a commit on top of your commit that gives you the block verifier factory and the initial block verifier func.

Reply:
I ended up putting the [commit](https://github.com/Fabric-orderer-collaboration/fabric/pull/18/commits/f5d234f80a0a79547025fdb1cf755b191f6b3b93) above on my other commit, please rebase on top :)